### PR TITLE
both of them finished, it is working but check needed

### PR DESCRIPTION
**Pull request by enesdemirelus**

### CODE REVIEW

File: web_project/blog/management/commands/create_ai_post.py
Comment (Guz-Ali) [inline web_project/blog/management/commands/create_ai_post.py:1]: create a new file called create_news_cards.py and put these in that. Revert back to original for create_ai_post.py .

File: web_project/blog/management/commands/create_ai_post.py [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
+from django.utils import timezone
 from blog.tasks import AINewsGenerator
+import os
+from django.utils import timezone
+
+def create_newscard(text_block):
+    news_items = newsParser(text_block)
+
+    for index, item in enumerate(news_items, start=1):
+        print(f"DEBUG: Processing news item {index} - Title: {item['title']}")
```
Comment (Guz-Ali) [inline web_project/blog/management/commands/create_ai_post.py:13]: remove debug prints

File: web_project/blog/tasks.py [hunk 2]
```diff
@@ -124,30 +124,29 @@ def generate_ai_text(self, prompt):
...
+                'text': f'create a header image for this news, make it realistic: {prompt}',
+            },
+            headers={'api-key': api_key}
         )
-        url = f"{r.json()['share_url']}"
         
-        if url:
-            output_dir = os.path.join(settings.MEDIA_ROOT, 'ai_images')
-            os.makedirs(output_dir, exist_ok=True)
+        if r.status_code == 200:
```
Comment (Guz-Ali) [inline web_project/blog/tasks.py:134]: check out fail early method for if-else statements. This allows to minimize nested if-else statements.
instead of 
```
if (right):
  do task
if (not right):
  do default / fail
```
do this
```
if (not right): 
  do default / fail
do task
```
check out the resource in here for more info:
https://docs.google.com/document/d/1QD6GLuVkEq1rIgCE5OMCSpP-IK-V6Q8kikACv-tlhAI/edit?tab=t.0

File: web_project/blog/utils.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import re
+from datetime import datetime
+
+def newsParser(text_block):
+    print("DEBUG: Full text block received:")
+    print(repr(text_block))
+
+    pattern = (
```
Comment (Guz-Ali) [inline web_project/blog/utils.py:8]: i see where GPT was used :) I was thinking of a simpler algorithm of just parsing based on new lines, but if this works and was tested thoroughly, it's fine.

File: web_project/blog/utils.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        date_str = match.group("date").strip()
+        content = match.group("content").strip()
+        link = match.group("link").strip()
+
+        try:
+            date = datetime.strptime(date_str, "%m/%d/%Y").date()
+        except ValueError:
+            date = None 
+            
+        news_items.append({
```
Comment (Guz-Ali) [inline web_project/blog/utils.py:32]: good algo

File: web_project/blog/management/commands/create_ai_post.py [hunk 5]
```diff
@@ -1,15 +1,50 @@
...
+
+        try:
+            # Temporary fix for image generation because we are out of credits.
```
Comment (Guz-Ali) [inline web_project/blog/management/commands/create_ai_post.py:16]: let's see.. Should we buy a premium account on DeepAI or search for something else?

File: web_project/blog/management/commands/create_ai_post.py [hunk 6]
```diff
@@ -1,15 +1,50 @@
...
+                title=item['title'],
+                content=item['content'],
+                date_posted=timezone.now(),
+                date_created=timezone.now(),
+                url_of_the_news=item['link'],
+                summary=item['content'][:500],
+                is_public=True
+            )
+
+            news_card.image = image_filename
```
Comment (Guz-Ali) [inline web_project/blog/management/commands/create_ai_post.py:32]: any reason why image is added later on?

File: web_project/blog/management/commands/create_ai_post.py (hunk 5 above)
Comment (enesdemirelus) [reply web_project/blog/management/commands/create_ai_post.py:16]: i guess something else would be better, this one creates very ugly images.

File: web_project/blog/management/commands/create_ai_post.py (hunk 6 above)
Comment (enesdemirelus) [reply web_project/blog/management/commands/create_ai_post.py:32]: I probably add the image later and forgot to put it inside of the dict. Fixed.

File: web_project/blog/utils.py (hunk 3 above)
Comment (enesdemirelus) [reply web_project/blog/utils.py:8]: Yes, I had to use gpt because it is extremely hard to parse the text coming from gemini. It could take days to do something like what gpt does in couple seconds.

File: web_project/blog/utils.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import re
+from datetime import datetime
+
+def newsParser(text_block):
```
Comment (Guz-Ali) [inline web_project/blog/utils.py:4]: python uses ```this_naming_convention``` for function names. quick fix

**Comment by Guz-Ali** [comment]
looks good. check out the last comment i made and it should be good to merge.
